### ISSUE TITLE
Failed to import some settings

### ISSUE THREAD

**Issue by ryancdotorg** [thunderbird/thunderbird-android#6689]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.508

### Where did you get the app from?

Google Play

### Android version

12 + 13

### Steps to reproduce

On old phone:

1. Settings
2. Export settings
3. Leave "General settings" and my account settings both checked
4. Export
5. Save exported settings
6. Transfer to new phone via "Nearby Share"

On new phone:

1. Import settings
2. Select file
3. Pick transferred settings
4. Leave "General settings" and my account settings both checked
5. Import

### Expected behavior

All settings including account imported.

### Actual behavior

Error message stating "Failed to import some settings" - "General settings" has a check, but my account settings shows an exclamation mark.

### Logs

Nothing relevant, but for completeness:

```
02-20 11:31:34.024 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:31:34.024 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:32:05.080 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:32:05.080 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:32:06.718 29139 29162 D OpenGLRenderer: endAllActiveAnimators on 0xb400007af4393000 (InsetDrawable) with handle 0xb400007a04368e90
02-20 11:32:07.559 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:32:07.559 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:35:51.824 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:35:51.824 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:35:53.253 29139 29162 D OpenGLRenderer: endAllActiveAnimators on 0xb400007af43b4650 (InsetDrawable) with handle 0xb400007a04377050
02-20 11:35:53.973 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:35:53.973 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:36:55.444 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:36:55.444 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
02-20 11:36:57.870 29139 29162 D OpenGLRenderer: endAllActiveAnimators on 0xb400007af43f9320 (InsetDrawable) with handle 0xb400007a0435ca40
02-20 11:36:58.263 29139 29139 W OnBackInvokedCallback: OnBackInvokedCallback is not enabled for the application.
02-20 11:36:58.263 29139 29139 W OnBackInvokedCallback: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
```

**Comment by cketti** [comment]
Can you please enable debug logging, import only the account settings, then export the debug log?

**Comment by ryancdotorg** [comment]
There is no way to enable debug logging without first configuring an account. If I set up the account manually

Here's the error, which is caused by one of my sending identities having an email address in the form of `"localpart@domain.example" <localpart@domain.example>`

```
02-20 11:42:58.393 29139 30927 E SettingsImporter: Encountered invalid setting while importing account "redacted@redacted.test"
02-20 11:42:58.393 29139 30927 E SettingsImporter: com.fsck.k9.preferences.Settings$InvalidSettingValueException
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.preferences.SettingsImporter.importIdentities(SettingsImporter.java:564)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.preferences.SettingsImporter.importAccount(SettingsImporter.java:455)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.preferences.SettingsImporter.importSettings(SettingsImporter.java:221)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.ui.settings.import.SettingsImportViewModel.importSettings(SettingsImportViewModel.kt:386)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.ui.settings.import.SettingsImportViewModel.access$importSettings(SettingsImportViewModel.kt:28)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at com.fsck.k9.ui.settings.import.SettingsImportViewModel$startImportSettings$1$1$1.invokeSuspend(SettingsImportViewModel.kt:303)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
02-20 11:42:58.393 29139 30927 E SettingsImporter:      at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

Editing that email address on the old phone, then re-exporting allows the import to complete.

**Comment by cketti** [comment]
The email address doesn't pass the validation in [EmailAddressValidator](https://github.com/thundernest/k-9/blob/7c928c4e8de167faef1aa511ebb500d374145430/app/core/src/main/java/com/fsck/k9/EmailAddressValidator.kt). Apparently we don't use this check when users create or edit an identity.

To end up with a "To" header value of `"localpart@domain.example" <localpart@domain.example>` use `localpart@domain.example` both under "Your name" and "Email address" in the "Edit identity" screen.

**Comment by fereshtehamiri** [comment]
@ryancdotorg 
Hello Dear,

I have been trying to reproduce the issue you described ((in Samsung A32, Android 13)), but I am having difficulty doing so. I was wondering if you could provide me with more information on the scenario you experienced in order to help us resolve the issue.

Specifically, could you please provide me with more details on how to reproduce the issue, as well as any other information that could be helpful in diagnosing and fixing the bug?

could you attach a picture or video for this scenario?

Your assistance in this matter would be greatly appreciated.

Thank you for your time

**Comment by ryancdotorg** [comment]
@fereshtehamiri Please re-read the previous comments on this issue, as well as the linked PR.

@cketti Should this issue be closed now?

**Comment by cketti** [comment]
The import code still needs to be fixed to skip importing invalid identities.

**Comment by fereshtehamiri** [comment]
@ryancdotorg 
hello 
Thank you for your explanation, but this bug did not occur for me (in Samsung A32, Android 13, K9 App Version: 6.702
) and All settings including account (username) imported.

**Comment by cketti** [comment]
@fereshtehamiri: This issue contains enough information to reproduce the bug:

1. Install K-9 Mail 6.508.
2. Create an identity with the value `"localpart@domain.example" <localpart@domain.example>` in the "email address" text field.
3. Export settings.
4. Install latest K-9 Mail version.
5. Try to import the settings exported in step 3.

PR #6696 changed the app so users can no longer create an identity with invalid values. I've created issue #6955 with what work is left to deal with this bug.

Closing this issue.
